bin/generate-groups-inventory.bash

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+
+here="$(dirname "$(readlink -f "$0")")"
+# shellcheck source=bin/common.bash
+source "${here}/common.bash"
+
+# shellcheck source=bin/common.bash
+source "${here}/inventory-parser.bash"
+
+if [ -e "${config[groups_inventory_file]}" ]; then 
+    log_info_no_newline "${config[groups_inventory_file]} will be overwritten, Proceed [y/N] ? "
+    read -r reply
```

Comment:
```suggestion
    read -n 1 -r reply
```
> `help read`
> -n nchars	return after reading NCHARS characters rather than waiting for a newline, but honor a delimiter if fewer than

so there's no chance of anyone typing the full 'yes' and you don't have to press Enter